dotnet/src/SemanticKernel.Skills/Skills.Compression/FileCompressionSkill.cs

```diff
@@ -0,0 +1,125 @@
+        /// </summary>
+        public const string DestinationDirectoryPath = "destinationDirectoryPath";
+
+        /// <summary>
+        /// File path where to save compressed data.
+        /// </summary>
+        public const string DestinationFilePath = "destinationFilePath";
+    }
+
+    private readonly IFileCompressor _fileCompressor;
+    private readonly ILogger<FileCompressionSkill> _logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FileCompressionSkill"/> class.
+    /// </summary>
+    /// <param name="fileCompressor">File compressor implementation</param>
+    /// <param name="logger">Optional logger</param>
+    public FileCompressionSkill(IFileCompressor fileCompressor, ILogger<FileCompressionSkill>? logger = null)
+    {
+        this._fileCompressor = fileCompressor ?? throw new ArgumentNullException(nameof(fileCompressor));
+        this._logger = logger ?? new NullLogger<FileCompressionSkill>();
+    }
+
+    [SKFunction("Compresses an input file to an output file")]
+    [SKFunctionInput(Description = "Path of file to compress")]
+    [SKFunctionContextParameter(Name = Parameters.DestinationFilePath, Description = "Path of compressed file to create")]
+    public Task CompressFileAsync(string sourceFilePath, SKContext context)
+    {
+        this._logger.LogDebug($"{nameof(CompressFileAsync)} got called");
+
+        if (!context.Variables.Get(Parameters.DestinationFilePath, out string destinationFilePath))
+        {
+            this._logger.LogError($"Missing context variable in {nameof(CompressFileAsync)}");
+            string errorMessage = $"Missing variable {Parameters.DestinationFilePath}";
+            context.Fail(errorMessage);
+
+            return Task.FromException(new KeyNotFoundException(errorMessage));
+        }
+
+        return this._fileCompressor.CompressFileAsync(sourceFilePath, destinationFilePath, context.CancellationToken);
+    }
+
+    [SKFunction("Compresses a directory to an output file")]
+    [SKFunctionInput(Description = "Path of directory to compress")]
+    [SKFunctionContextParameter(Name = Parameters.DestinationFilePath, Description = "Path of compressed file to create")]
+    public Task CompressDirectoryAsync(string sourceDirectoryPath, SKContext context)
+    {
+        this._logger.LogDebug($"{nameof(CompressDirectoryAsync)} got called");
+
+        if (!context.Variables.Get(Parameters.DestinationFilePath, out string destinationFilePath))
+        {
+            this._logger.LogError($"Missing context variable in {nameof(CompressDirectoryAsync)}");
+            string errorMessage = $"Missing variable {Parameters.DestinationFilePath}";
+            context.Fail(errorMessage);
+
+            return Task.FromException(new KeyNotFoundException(errorMessage));
+        }
+
+        return this._fileCompressor.CompressDirectoryAsync(sourceDirectoryPath, destinationFilePath, context.CancellationToken);
```

Comment:
what's the name of the file created? what happens if the file already exists, e.g. we retry the Plan and the same folder is zipped multiple times?

Reply:
The name of the file created is passed as destinationFilePath.

If the file exists already, zip file creation fails with an error like the following:

This fails on attempted overwrites with
Exception thrown: 'System.IO.IOException' in System.Private.CoreLib.dll
An exception of type 'System.IO.IOException' occurred in System.Private.CoreLib.dll but was not handled in user code
The file 'C:\Users\gillahaye\AppData\Local\Temp\biorgmb1.ojd.zip' already exists.